[Bug] Qadence protocol ZNE test fails, requires SimConfig updates

Pulser backend written in qadence is using the following to create sim_config. Pulser has moved from using "_prob"  to "_rate" in their SimConfigs.  
`sim_config = {"noise": noise.protocol, noise.protocol + "_prob": noise_prob}`

The tests (ZNE) in qadence protocols are failing due to this with the following error

```
FAILED tests/test_zne.py::test_analog_zne_with_noise_levels[analog_block0-observable0-noise_probs0-depolarizing] - TypeError: SimConfig.__init__() got an unexpected keyword argument 'depolarizing_prob'
FAILED tests/test_zne.py::test_analog_zne_with_noise_levels[analog_block1-observable1-noise_probs1-dephasing] - TypeError: SimConfig.__init__() got an unexpected keyword argument 'dephasing_prob'
FAILED tests/test_zne.py::test_analog_zne_with_pulse_stretching[analog_block0-observable0-noise_probs0-depolarizing-param_values0] - TypeError: SimConfig.__init__() got an unexpected keyword argument 'depolarizing_prob'
```

Not sure if this is valid, but I think there should be feature tests included that fail when backend changes happen, so that wrappers written on it run smoothly.

--- a/qadence/backends/pulser/backend.py
+++ b/qadence/backends/pulser/backend.py
@@ -243,7 +243,7 @@ class Backend(BackendInterface):
         # Pulser requires numpy types.
         for noise_prob in noise_probs.numpy():
             batched_dm = []
-            sim_config = {"noise": noise.protocol, noise.protocol + "_prob": noise_prob}
+            sim_config = {"noise": noise.protocol, noise.protocol + "_rate": noise_prob}
             self.config.sim_config = SimConfig(**sim_config)
 
             for i, param_values_el in enumerate(vals):

